cloudinit/cmd/main.py

```diff
@@ -202,22 +203,35 @@ def attempt_cmdline_url(path, network=True, cmdline=None):
     header = b"#cloud-config"
     try:
         resp = url_helper.read_file_or_url(**kwargs)
+        sniffed_content = b""
         if resp.ok():
-            data = resp.contents
-            if not resp.contents.startswith(header):
+            is_cloud_cfg = True
+            if isinstance(resp, url_helper.UrlResponse):
+                try:
+                    sniffed_content += next(
+                        resp.iter_content(chunk_size=len(header))
+                    )
+                except StopIteration:
+                    pass
+                if not sniffed_content.startswith(header):
+                    is_cloud_cfg = False
+            elif not resp.contents.startswith(header):
+                is_cloud_cfg = False
+            if not is_cloud_cfg:
```

Comment:
Please add a DEPRECATED: warning to `parse_cmdline_url` if "url" to suggest using cloud-config-url instead if the cmdline param used detected as we generally don't want people providing such a generic kernel cmdline key which is used by other system init solutions if we can avoid it.
```suggestion
            if is_cloud_cfg:
                if cmdline_name == "url"
                    LOG.warning("DEPRECATION: `url` kernel commandline key is deprecated for providing cloud-config via URL. Please use `cloud-config-url` kernel command line parameter instead")
            else:
```

Reply:
done